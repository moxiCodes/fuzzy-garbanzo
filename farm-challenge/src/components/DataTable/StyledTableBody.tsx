import { Checkbox, TableBody, TableCell, TableRow } from '@mui/material'

const tableRowSx = {
  '&:nth-of-type(even)': {
    backgroundColor: '#EEEEEE',
  },
}
const tableImgStyles = { width: '50px', borderRadius: '10px' }

const StyledTableBody = ({ rows }: { rows: Record<any, any>[] }) => {
  return (
    <TableBody>
      {rows.map((row: Record<any, any>) => (
        <TableRow key={row.name} sx={tableRowSx}>
          <TableCell align="left">
            <img
              alt={row.animalType}
              src={`./${process.env.PUBLIC_URL}/animals/${row.animalType}.png`}
              style={tableImgStyles}
            />
          </TableCell>
          <TableCell align="left">{row.name}</TableCell>
          <TableCell align="left">{row.age}</TableCell>
          <TableCell align="left">{row.gender}</TableCell>
          <TableCell align="left">{row.animalType}</TableCell>
          <TableCell align="left">{row.worth}</TableCell>
          <TableCell align="left">
            <Checkbox checked={row.isAlive} disabled aria-disabled />
          </TableCell>
        </TableRow>
      ))}
    </TableBody>
  )
}

export default StyledTableBody
