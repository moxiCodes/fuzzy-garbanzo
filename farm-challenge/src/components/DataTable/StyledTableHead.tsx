import { TableCell, TableHead, TableRow, Typography } from '@mui/material'

const StyledTableHead = ({ columns }: { columns: string[] }) => {
  return (
    <TableHead sx={{ backgroundColor: '#EEEEEE' }}>
      <TableRow>
        {columns.map((heading: string) => (
          <TableCell key={heading} align="left">
            <Typography variant="overline">{heading}</Typography>
          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  )
}

export default StyledTableHead
