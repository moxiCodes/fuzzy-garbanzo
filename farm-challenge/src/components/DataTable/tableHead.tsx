import { TableCell, TableHead, TableRow, Typography } from '@mui/material'

const columns = [
  'HeadShot',
  'Name',
  'Age',
  'Pronouns',
  'Type',
  'Worth',
  'Angel Status',
]
const StyledTableHead = () => {
  return (
    <TableHead sx={{ backgroundColor: '#EEEEEE' }}>
      <TableRow>
        {columns.map((heading: string) => (
          <TableCell key={heading} align="left">
            <Typography variant="overline">{heading}</Typography>
          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  )
}

export default StyledTableHead
