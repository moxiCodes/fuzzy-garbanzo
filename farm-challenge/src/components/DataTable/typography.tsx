import { Typography } from '@mui/material'
import { ReactNode } from 'react'

const StyledTypography = ({ children }: { children: ReactNode }) => {
  return (
    <Typography
      variant="subtitle1"
      gutterBottom
      sx={{ marginBottom: 1.2, color: '#666666' }}
    >
      {children}
    </Typography>
  )
}

export default StyledTypography
