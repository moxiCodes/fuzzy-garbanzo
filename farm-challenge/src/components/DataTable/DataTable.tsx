import { TableContainer, Table, Paper, Container } from '@mui/material'

import StyledTypography from './typography'
import StyledTableBody from './tableBody'
import StyledTableHead from './tableHead'
import { AnimalData } from '../../types/types'

const DataTable = ({ data }: { data: AnimalData[] }) => {
  return data?.length < 1 ? (
    <StyledTypography>NAE ANIMALS FOUND WITH THAT CRITERIA...</StyledTypography>
  ) : (
    <Container disableGutters>
      <StyledTypography>Your Animals</StyledTypography>
      <TableContainer component={Paper} variant="outlined">
        <Table sx={{ minWidth: 650 }}>
          <StyledTableHead />
          <StyledTableBody rows={data} />
        </Table>
      </TableContainer>
    </Container>
  )
}
export default DataTable
