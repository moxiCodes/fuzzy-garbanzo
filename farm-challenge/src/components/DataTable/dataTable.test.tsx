import { render } from '@testing-library/react'
import { AnimalData } from '../../types/types'
import DataTable from './dataTable'

const data = [
  {
    age: 10,
    animalType: 'cow',
    gender: 'female',
    isAlive: false,
    name: 'Adriana Burgess',
    worth: 43.87,
    _id: '6690e34e1cb6c589294bf1e9',
  },
  {
    age: 6,
    animalType: 'cow',
    gender: 'male',
    isAlive: true,
    name: 'Aguirre Andrews',
    worth: 10.4,
    _id: '6690e34eb1965a30baab8221',
  },
  {
    age: 9,
    animalType: 'cow',
    gender: 'female',
    isAlive: true,
    name: 'Alejandra Burnett',
    worth: 42.7,
    _id: '6690e34ecd191888af125a38',
  },
] as AnimalData[]

describe('Displays table of animals or a message', () => {
  it('displays a message if no data is found', async () => {
    expect(render(<DataTable data={[]} />)).toMatchSnapshot()
  })
  it('displays a table if data exists', async () => {
    expect(render(<DataTable data={data} />)).toMatchSnapshot()
  })
})
