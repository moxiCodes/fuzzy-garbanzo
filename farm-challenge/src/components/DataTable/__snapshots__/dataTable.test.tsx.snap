// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays table of animals or a message displays a message if no data is found 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1gxqp38-MuiTypography-root"
      >
        NAE ANIMALS FOUND WITH THAT CRITERIA...
      </h6>
    </div>
  </body>,
  "container": <div>
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1gxqp38-MuiTypography-root"
    >
      NAE ANIMALS FOUND WITH THAT CRITERIA...
    </h6>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Displays table of animals or a message displays a table if data exists 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-1p4tas-MuiContainer-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1gxqp38-MuiTypography-root"
        >
          Your Animals
        </h6>
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiTableContainer-root css-h2cton-MuiPaper-root-MuiTableContainer-root"
        >
          <table
            class="MuiTable-root css-quj9j6-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-q0r4vt-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  <span
                    class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                  >
                    HeadShot
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  <span
                    class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                  >
                    Name
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  <span
                    class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                  >
                    Age
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  <span
                    class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                  >
                    Pronouns
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  <span
                    class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                  >
                    Type
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  <span
                    class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                  >
                    Worth
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  <span
                    class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                  >
                    Angel Status
                  </span>
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root css-sjb7q1-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <img
                    alt="cow"
                    src=".//animals/cow.png"
                    style="width: 24px; border-radius: 5px;"
                  />
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  Adriana Burgess
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  10
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  female
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  cow
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  43.87
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <span
                    aria-disabled="true"
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-disabled PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-disabled MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    tabindex="-1"
                  >
                    <input
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate="false"
                      disabled=""
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                </td>
              </tr>
              <tr
                class="MuiTableRow-root css-sjb7q1-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <img
                    alt="cow"
                    src=".//animals/cow.png"
                    style="width: 24px; border-radius: 5px;"
                  />
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  Aguirre Andrews
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  6
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  male
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  cow
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  10.4
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <span
                    aria-disabled="true"
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-disabled PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked Mui-disabled MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    tabindex="-1"
                  >
                    <input
                      checked=""
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate="false"
                      disabled=""
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                      />
                    </svg>
                  </span>
                </td>
              </tr>
              <tr
                class="MuiTableRow-root css-sjb7q1-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <img
                    alt="cow"
                    src=".//animals/cow.png"
                    style="width: 24px; border-radius: 5px;"
                  />
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  Alejandra Burnett
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  9
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  female
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  cow
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  42.7
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <span
                    aria-disabled="true"
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-disabled PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked Mui-disabled MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    tabindex="-1"
                  >
                    <input
                      checked=""
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate="false"
                      disabled=""
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                      />
                    </svg>
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-1p4tas-MuiContainer-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1gxqp38-MuiTypography-root"
      >
        Your Animals
      </h6>
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiTableContainer-root css-h2cton-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-quj9j6-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-q0r4vt-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                >
                  HeadShot
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                >
                  Name
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                >
                  Age
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                >
                  Pronouns
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                >
                  Type
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                >
                  Worth
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
                >
                  Angel Status
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-sjb7q1-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <img
                  alt="cow"
                  src=".//animals/cow.png"
                  style="width: 24px; border-radius: 5px;"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Adriana Burgess
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                10
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                female
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                cow
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                43.87
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <span
                  aria-disabled="true"
                  class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-disabled PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-disabled MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                  tabindex="-1"
                >
                  <input
                    class="PrivateSwitchBase-input css-1m9pwf3"
                    data-indeterminate="false"
                    disabled=""
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckBoxOutlineBlankIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-sjb7q1-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <img
                  alt="cow"
                  src=".//animals/cow.png"
                  style="width: 24px; border-radius: 5px;"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Aguirre Andrews
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                6
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                male
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                cow
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                10.4
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <span
                  aria-disabled="true"
                  class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-disabled PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked Mui-disabled MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                  tabindex="-1"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input css-1m9pwf3"
                    data-indeterminate="false"
                    disabled=""
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckBoxIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-sjb7q1-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <img
                  alt="cow"
                  src=".//animals/cow.png"
                  style="width: 24px; border-radius: 5px;"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Alejandra Burnett
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                9
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                female
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                cow
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                42.7
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <span
                  aria-disabled="true"
                  class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-disabled PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked Mui-disabled MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                  tabindex="-1"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input css-1m9pwf3"
                    data-indeterminate="false"
                    disabled=""
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckBoxIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
